src/Aspire.Hosting/Extensions/ResourceBuilderExtensions.cs

```diff
@@ -228,6 +228,38 @@ public static IResourceBuilder<TDestination> WithReference<TDestination>(this IR
         });
     }
 
+    /// <summary>
+    /// Injects a connection string as an environment variable. The format of the environment variable will be "ConnectionStrings__{name}={value}." If the
+    /// connection string is not specified, the configuration system will be queried for a connection string using the connection string name.
+    /// </summary>
+    /// <typeparam name="TDestination"></typeparam>
+    /// <param name="builder">The resource where connection string will be injected.</param>
+    /// <param name="connectionString">A connection string</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{TDestination}"/>.</returns>
+    public static IResourceBuilder<TDestination> WithReference<TDestination>(this IResourceBuilder<TDestination> builder, ConnectionString connectionString)
```

Comment:
Won't block for preview 2 but need to figure out how this works with manifests where we want to request the connection string from the user/environment at deployment time.